src/Microsoft.Tye.Core/ConfigModel/ConfigFactory.cs

```diff
@@ -97,5 +98,8 @@ private static ConfigApplication FromYaml(FileInfo file)
             using var parser = new YamlParser(file);
             return parser.ParseConfigApplication();
         }
+
+        private static string Normalization(string name)
+            => Regex.Replace(name.ToLowerInvariant(), "[^0-9A-Za-z-]+", "-");
```

Comment:
ex : "@some.proj!.csproj". it wont be ideal case. with this regex results :

>serivce name =  -some-proj-

isn't ok..

Reply:
There will always be cases that break, it's a heuristic. Have you seen a project called `@some.proj!.csproj` before? 😆

Reply:
Before also replacing “.” Is missing. Anyway doubt has clarified Thanks 😁..

Reply:
Everything fine??🙂